game-app/game-core/src/main/java/games/strategy/triplea/ui/panel/move/MovePanel.java

```diff
@@ -158,63 +158,45 @@ public void unitsSelected(
 
         private void selectUnitsToMove(
             final List<Unit> units, final Territory t, final MouseDetails mouseDetails) {
-          // are any of the units ours, note - if no units selected that's still ok
-          if (!BaseEditDelegate.getEditMode(getData().getProperties())
-              || !selectedUnits.isEmpty()) {
-            for (final Unit unit : units) {
-              if (!unit.getOwner().equals(getUnitOwner(selectedUnits))) {
-                return;
-              }
-            }
+          final boolean editMode = BaseEditDelegate.getEditMode(getData().getProperties());
+
+          GamePlayer requiredOwner = null;
+          if (!editMode) {
+            requiredOwner = getCurrentPlayer();
+          } else if (!selectedUnits.isEmpty()) {
+            requiredOwner = selectedUnits.iterator().next().getOwner();
           }
+          if (!units.stream().allMatch(Matches.unitIsOwnedBy(requiredOwner))) {
+            return;
+          }
+
           // basic match criteria only
           final Predicate<Unit> unitsToMoveMatch = getMovableMatch(null, null);
-          final Predicate<Collection<Unit>> ownerMatch =
-              unitsToCheck -> {
-                final GamePlayer owner = unitsToCheck.iterator().next().getOwner();
-                for (final Unit unit : unitsToCheck) {
-                  if (!owner.equals(unit.getOwner())) {
-                    return false;
-                  }
-                }
-                return true;
-              };
           if (units.isEmpty() && selectedUnits.isEmpty() && !mouseDetails.isShiftDown()) {
             final List<Unit> unitsToMove = t.getUnitCollection().getMatches(unitsToMoveMatch);
             if (unitsToMove.isEmpty()) {
               return;
             }
-            final String text = "Select units to move from " + t.getName();
-            final UnitChooser chooser;
-            if (BaseEditDelegate.getEditMode(getData().getProperties())
-                && !CollectionUtils.getMatches(
-                        unitsToMove, Matches.unitIsOwnedBy(getUnitOwner(unitsToMove)))
-                    .containsAll(unitsToMove)) {
-              // use matcher to prevent units of different owners being chosen
-              chooser =
-                  new UnitChooser(
-                      unitsToMove,
-                      selectedUnits,
-                      null,
-                      UnitSeparator.SeparatorCategories.builder().build(),
-                      false,
-                      getMap().getUiContext(),
-                      ownerMatch);
-            } else {
-              chooser =
-                  new UnitChooser(
-                      unitsToMove,
-                      selectedUnits,
-                      null,
-                      UnitSeparator.SeparatorCategories.builder().build(),
-                      false,
-                      getMap().getUiContext());
```

Comment:
We didn't really need to have two branches. The edit mode branch added a filter that ensured only units of the same player were selected. But this is fine to do in non-edit mode too, though it's a no-op.